### ISSUE TITLE
Feature Request: Flagging Species for ATS-Transport to Ignore (Application: Kinetic Sorption)

### ISSUE THREAD

**Issue by saubhagya-gatech** [amanzi/ats#93]
Dear Developers,

This is to request a feature in ATS that can tell ATS to not transport certain components. Ignoring these components for transport will enable implementation of kinetic sorption conveniently. 

Thanks
Saubhagya

**Comment by kchenhyd** [comment]
Dear developers,

I want to simulate microbial growth/death in microbial reactions, which requires the setting of immobile species. It would be great if ATS could support not-transport capability.

Thanks,
Kewei

**Comment by ecoon** [comment]
This is ready for revisiting in ecoon/ats_transport_refactor.  

@saubhagya-gatech is this still of interest/useful?  Is the goal really just: don't advect some components, even though they are "aqueous" components from a geochemical engine perspective?  Note that Alquimia currently has a mechanism for sorption, with a separate "sorbed_quantity" for each component.  Is this not what you need?

**Comment by smolins** [comment]
I think this would be useful beyond the application described by
@saubhagya-gatech <https://github.com/saubhagya-gatech>. Also, PFLOTRAN has
a capability for immobile species that some users want to use. I would not
say this is of high priority, though. Maybe polling the users group to
gauge interest?

**Comment by saubhagya-gatech** [comment]
Hi @ecoon  @smolins 

> Is the goal really just: don't advect some components, even though they are "aqueous" components from a geochemical engine perspective?

^ This was to mimic kinetic sorption, where we still define C_sorb as aq component in PFLOTRAN, just not advect in transport PK and also exclude from aq. reactions in PFLOTRAN. At that time, only equilibrium sorption was supported in PFLOTRAN, hence this is a work around. 

Having actual sorbed quantities for each component would be ideal and cleaner if possible. In PFLOTRAN, sorbed quantities do not partake in reactions except radioactive decay (happens in both aq. and sorbed phase).

I see Sergi's point as well. There could be mechanisms other than sorption that makes component immobile, and their participation in the reaction network looks different than sorbed quantities?

**Comment by ecoon** [comment]
Ok, it's a good time to add this because of the transport refactor, and should be pretty easy to do.

Should they still experience molecular diffusion, or skip that as well?  If they're not advecting, clearly then no dispersion I assume?

**Comment by saubhagya-gatech** [comment]
Yes, no advection and dispersion. I think they should not experience molecular diffusion since molecular diffusion refers to the movement of molecules due to concentration gradients within a fluid phase (gas or liquid) (governed by Fick’s laws and requires molecular mobility, which occurs only in phases where molecules are free to move), and sorbed quantities are immobile. However, sorption affects apparent/emergent diffusion behavior as it affects amount of specie that is mobile and available for diffusion. For equilibrium sorption, this shows up as retardation factor in diffusive flux term.

**Comment by smolins** [comment]
Beyond this, having solutes not subject to advection-dispersion-diffusion
processes may offer occasional flexible options with numerical
advantages for reaction models. Reaction models typically use aqueous
concentrations of primary species as primary unknowns. Other things that
are not transported tend to be secondary unknowns and/or updated after the
step. It may be convenient sometimes, e.g. in stiff problems, to use
aqueous concentrations to represent them but prevent them from being
transported. An example would be microbial biomass, even if immobile, it
may be better captured with aqueous concentrations.
